modules/compatibility/src/test/java/org/apache/ignite/compatibility/sql/SqlQueryRegressionsTest.java

```diff
@@ -0,0 +1,359 @@
+import org.apache.ignite.spi.discovery.tcp.ipfinder.vm.TcpDiscoveryVmIpFinder;
+import org.apache.ignite.testframework.junits.multijvm.IgniteProcessProxy;
+import org.junit.Test;
+
+/**
+ * Test for SQL queries regressions detection.
+ * It happens in the next way:
+ * 
+ * 1. Test starts two different Ignite versions: current version and the old one.
+ * 2. Then framework executes (randomly chosen/generated) equivalent queries in both versions.
+ * 3. Execution time for both version is measured and if it exceeds some threshold, the query marked as suspected.
+ * 4. All suspected queries are submitted to both Ignite versions one more time to get rid of outliers.
+ * 5. If a poor execution time is reproducible for suspected query,
+ *    this query is reported as a problematic and test fails because of it.
+ */
+@SuppressWarnings("TypeMayBeWeakened")
+public class SqlQueryRegressionsTest extends IgniteCompatibilityAbstractTest {
+    /**
+     * You may wanted to set this seed to particular value in case you will need to troubleshoot particular run. If it
+     * set to the some value (not {@code null}), it will be used in random generator, otherwise generator will be
+     * initialized with random seed.
+     */
+    private static final Integer SEED = null;
+
+    /** */
+    private static final String BASE_VERSION_PARAM = "BASE_VERSION";
+
+    /** */
+    private static final String BASE_IS_IGNITE_PARAM = "BASE_IS_IGNITE";
+
+    /** */
+    private static final String SEED_PARAM = "SEED";
+
+    /** */
+    private static final String DEFAULT_BASE_VERSION = "8.7.22";
+
+    /** */
+    private static final boolean DEFAULT_BASE_IS_IGNITE = false;
+
+    /** */
+    private static final int OLD_JDBC_PORT = 10800;
+
+    /** */
+    private static final int NEW_JDBC_PORT = 10802;
+
+    /** */
+    private static final long BENCHMARK_TIMEOUT = 60_000;
+
+    /** */
+    private static final long WARM_UP_TIMEOUT = 5_000;
+
+    /** */
+    private static final String JDBC_URL = "jdbc:ignite:thin://127.0.0.1:";
+
+    /** */
+    private static final TcpDiscoveryIpFinder OLD_VER_FINDER = new TcpDiscoveryVmIpFinder(true) {{
+        setAddresses(Collections.singleton("127.0.0.1:47500..47509"));
+    }};
+
+    /**  */
+    private static final TcpDiscoveryVmIpFinder NEW_VER_FINDER = new TcpDiscoveryVmIpFinder(true) {{
+        setAddresses(Collections.singleton("127.0.0.1:47510..47519"));
+    }};
+
+    /** Model factories. */
+    private static final List<ModelFactory> MODEL_FACTORIES = Arrays.asList(
+        new Person.Factory(),
+        new Department.Factory(),
+        new Country.Factory(),
+        new City.Factory(),
+        new Company.Factory()
+    );
+
+    /** */
+    private static final Schema SCHEMA = new Schema();
+
+    static {
+        MODEL_FACTORIES.stream().map(ModelFactory::queryEntity).map(Table::new).forEach(SCHEMA::addTable);
+    }
+
+    /** */
+    private boolean baseIsIgnite;
```

Comment:
{@true} if base is Ignite, {@false} if it is GG ?

Reply:
Yep. Will note this in javadoc